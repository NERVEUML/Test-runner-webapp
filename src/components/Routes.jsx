/* /src/components/Routes.jsx  */

import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch
} from 'react-router-dom';

// Components
import Home from './Views/Home.jsx'; 
import ConfigurationList from './Views/ConfigurationList.jsx'; 
import EvaluationList from './Views/EvaluationList.jsx'; 

class Routes extends Component {
  render() {
    return (
      <Router>
        <div>
          <Switch>
            <Route exact path="/" component={ Home } />
            <Route exact path="/Configurations" component={ ConfigurationList } />
            <Route exact path="/Evaluations" component={ EvaluationList } />
            <Route path="*" render={() => <h1>404 NOT FOUND</h1>} />
          </Switch>
        </div>
      </Router>
    );
  }
}

export default Routes;