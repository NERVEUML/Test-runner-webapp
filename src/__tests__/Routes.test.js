/* /src/__tests__/Routes.test.js  */

//Libraries
import React from 'react';
import { shallow } from 'enzyme'
//Component To Be Tested
import Routes from '../components/Routes.jsx';

//Call to create a Shallow rendering of component
const wrapper = shallow(<Routes />)

describe('Routes Component', () => {
  test('render', () => {
    expect(wrapper.exists()).toBe(true);
    //The Routes Must be identical for the site to work 
    expect(wrapper).toMatchSnapshot();
  })
});