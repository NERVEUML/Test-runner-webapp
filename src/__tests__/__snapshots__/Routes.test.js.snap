// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes Component render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <BrowserRouter>
    <div>
        <Switch>
            <Route
                component={[Function]}
                exact={true}
                path="/"
            />
            <Route
                component={[Function]}
                exact={true}
                path="/Configurations"
            />
            <Route
                component={[Function]}
                exact={true}
                path="/Evaluations"
            />
            <Route
                path="*"
                render={[Function]}
            />
        </Switch>
    </div>
</BrowserRouter>,
  "nodes": Array [
    <BrowserRouter>
      <div>
            <Switch>
                  <Route
                        component={[Function]}
                        exact={true}
                        path="/"
                  />
                  <Route
                        component={[Function]}
                        exact={true}
                        path="/Configurations"
                  />
                  <Route
                        component={[Function]}
                        exact={true}
                        path="/Evaluations"
                  />
                  <Route
                        path="*"
                        render={[Function]}
                  />
            </Switch>
      </div>
</BrowserRouter>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Routes />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Routes {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <BrowserRouter>
          <div>
                    <Switch>
                              <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/"
                              />
                              <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/Configurations"
                              />
                              <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/Evaluations"
                              />
                              <Route
                                        path="*"
                                        render={[Function]}
                              />
                    </Switch>
          </div>
</BrowserRouter>,
        "_debugID": 2,
        "_renderedOutput": <BrowserRouter>
          <div>
                    <Switch>
                              <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/"
                              />
                              <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/Configurations"
                              />
                              <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/Evaluations"
                              />
                              <Route
                                        path="*"
                                        render={[Function]}
                              />
                    </Switch>
          </div>
</BrowserRouter>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Routes />,
}
`;
